doc/source/development.rst

```diff
@@ -99,6 +99,50 @@ directory will take effect without reinstalling the package.
 
 .. warning:: if you run ``python setup.py install``, files will be copied from the Ray directory to a directory of Python packages (``/lib/python3.6/site-packages/ray``). This means that changes you make to files in the Ray directory will not have any effect.
 
+Building Ray on Windows (full)
+------------------------------
+
+**Requirements**
+
+The following links were correct during the writing of this section. In case the URLs changed, search at the organizations' sites.
+
+- bazel 3.2
+- Microsoft Visual Studio 2019 (or Microsoft Build Tools 2019 - https://visualstudio.microsoft.com/downloads/#build-tools-for-visual-studio-2019)
+- JDK 15 (https://www.oracle.com/java/technologies/javase-jdk15-downloads.html)
+- Miniconda 3 (https://docs.conda.io/en/latest/miniconda.html)
+- git for Windows, version 2.31.1 or later (https://git-scm.com/download/win)
+
+**Steps**
+
+1. Enable Developer mode on Windows 10 systems. This is necessary so git can create symlinks.
+
+   1. Open Settings app;
+   2. Go to "Update & Security";
+   3. Go to "For Developers" on the left pane;
+   4. Turn on "Developer mode".
+
+2. Add the following Miniconda subdirectories to PATH. If Miniconda was installed for all users, the following paths are correct. If Miniconda is installed for a single user, adjust the paths accordingly.
+
+   - ``C:\ProgramData\Miniconda3``
+   - ``C:\ProgramData\Miniconda3\Scripts``
+   - ``C:\ProgramData\Miniconda3\Library\bin``
+
+3. Define an environment variable BAZEL_SH to point to bash.exe. If git for Windows was installed for all users, bash's path should be ``C:\Program Files\Git\bin\bash.exe``. If git was installed for a single user, adjust the path accordingly.
+
+4. Install cython:
+
+.. code-block:: shell
+
+  pip install cython
+
+5. Download ray source code and build it.
+
+.. code-block:: shell
+
+  # cd to the directory under which the ray source tree will be downloaded.
+  git clone -c core.symlinks=true https://github.com/ray-project/ray.git
+  cd ray\python
+  pip install -e . --verbose
 
 Fast, Debug, and Optimized Builds
```

Comment:
We should change this headline to a `------------------------------` headline (up one level), otherwise it will be a subsection of the windows section, which doesn't really make sense (it applies to both windows and linux).